### ISSUE TITLE
no-shadowed-variable false-positive for nested block scopes

### ISSUE THREAD

**Issue by paulmedynski** [palantir/tslint#4261]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.1.3
- __Running TSLint via__: brunch

#### TypeScript code being linted

```ts
{
  const foo = 0;
}
const foo = 1;
```

with `tslint.json` configuration:

```json
{
  "extends": "tslint:recommended"
}
```

#### Actual behavior

`ERROR: test.ts[2, 9]: Shadowed name: 'foo'`

#### Expected behavior

Variables in a deeper scope never shadow those in a shallower scope.  Is TSLint incorrectly assuming that the outer `foo` is being hoisted like `var` would do?  `const` doesn't hoist, so the outer `foo` doesn't exist until the inner `foo`'s block is done.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
